File: application/src/main/java/pl/mpakula/shorturl/application/config/AppConfig.java
package pl.mpakula.shorturl.application.config;

import org.apache.commons.validator.routines.UrlValidator;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpStatus;
import pl.mpakula.shorturl.core.UrlFacade;
import pl.mpakula.shorturl.core.ports.outgoing.UrlMappingRepository;

import java.util.Map;

@Configuration
@ComponentScan(basePackages = {"pl.mpakula.shorturl.infrastructure"})
class AppConfig {

    @Bean
    Map<HttpStatus, String> errorPages() {
        return Map.of(HttpStatus.NOT_FOUND, "error-404", HttpStatus.UNPROCESSABLE_ENTITY, "error-422");
    }

    @Bean
    UrlFacade urlFacade(@Value("${url.base_path}") String basePath, UrlMappingRepository urlMappingRepository) {
        return new UrlFacade(UrlValidator.getInstance(), basePath, urlMappingRepository);
    }

}

File: application/src/main/kotlin/pl/mpakula/shorturl/application/config/AppConfig.kt
package pl.mpakula.shorturl.application.config

import org.apache.commons.validator.routines.UrlValidator
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpStatus
import pl.mpakula.shorturl.core.UrlFacade
import pl.mpakula.shorturl.core.ports.outgoing.UrlMappingRepository

@Configuration
@ComponentScan(basePackages = ["pl.mpakula.shorturl.infrastructure"])
internal open class AppConfig {

    @Bean
    open fun errorPages(): Map<HttpStatus, String> =
        mapOf(HttpStatus.NOT_FOUND to "error-404", HttpStatus.UNPROCESSABLE_ENTITY to "error-422")

    @Bean
    open fun urlFacade(
        @Value("\${url.base_path}") basePath: String,
        urlMappingRepository: UrlMappingRepository
    ): UrlFacade = UrlFacade(UrlValidator.getInstance(), basePath, urlMappingRepository)

}